### PR TITLE
fixed the siutation when a user pass only a image

### PR DESCRIPTION
**Pull request by ChristofferEdlund**
# Problem
One should be able to user the albumentations transform in this way:
```

transform = AlbumentationsTransform(transform_compse())
img_tensor = transform(img)

```

Right now the transform output's a (img, target) tuple even when only an image is passed.

# Solution

Add a check if there is only an image as input, and return only image if that is the case.

# Changelog
- albumentation transform now returns only img and not (img, target) when only an image is present as input

### CODE REVIEW

File: tests/darwin/exporter/formats/export_darwin_1_0_test.py
Comment (dorfmanrobert) [inline tests/darwin/exporter/formats/export_darwin_1_0_test.py:1]: side question regarding these tests (do know this isn't part of the PR): isn't "paths" the new "complex_polygon"?

File: darwin/torch/transforms.py [hunk 1]
```diff
@@ -332,6 +340,9 @@ def __call__(self, image, annotation: dict = None) -> tuple:
         transformed_data = self.transform(**albu_data)
         image, transformed_annotation = self._post_process(transformed_data, annotation)
 
+        if len(annotation) < 1:
+            return image
```
[the remark this answers is not in the document]
Comment (ChristofferEdlund) [reply darwin/torch/transforms.py:344]: It is how other transforms seems to be called.


    def __getitem__(self, index: int) -> Tuple[Tensor, Tensor]:
        """
        See superclass for documentation.

        Parameters
        ----------
        index : int
            The index of the image.

        Returns
        -------
        Tuple[Tensor, Tensor]
            A tuple of tensors, where the first value is the image tensor and the second is the
            target's tensor.
        """
        img: PILImage.Image = self.get_image(index)
        if self.transform is not None:
            img_tensor = self.transform(img)
        else:
            img_tensor = to_tensor(img)

        target = self.get_target(index)

        return img_tensor, target


https://github.com/v7labs/darwin-py/blob/5b7ad8b4064eac4672cbe950f40855c2be2ea009/darwin/torch/dataset.py#L113C10-L113C10

So lets enable that behavior in our AlbumentationsTransform class as well.

[the remark this answers is not in the document]
Comment (dorfmanrobert) [reply darwin/torch/transforms.py:344]: maybe could default the value to None and use that instead of len() > 1?

File: tests/darwin/exporter/formats/export_darwin_1_0_test.py
Comment (dorfmanrobert) [reply tests/darwin/exporter/formats/export_darwin_1_0_test.py:1]: gotcha thanks

**Review by dorfmanrobert — APPROVED** [review]

File: darwin/torch/transforms.py [hunk 2]
```diff
@@ -349,7 +363,7 @@ def _pre_process(self, image: np.ndarray, annotation: dict) -> dict:
             albumentation_dict["labels"] = labels.tolist()
 
         masks = annotation.get("masks")
-        if masks is not None:
+        if masks is not None and masks.numel() > 0:  # using numel() to check if tensor is non-empty
```
Comment (dorfmanrobert) [inline darwin/torch/transforms.py:366]: if masks is None this will error

File: darwin/torch/transforms.py [hunk 3]
```diff
@@ -364,8 +378,15 @@ def _post_process(self, albumentation_output: dict, annotation: dict) -> tuple:
         bboxes = albumentation_output.get("bboxes")
         if bboxes is not None:
             output_annotation["boxes"] = torch.tensor(bboxes)
+
             if "area" in annotation and "masks" not in albumentation_output:
```
Comment (dorfmanrobert) [inline darwin/torch/transforms.py:382]: just wondering, why is it necessary to check for "masks" not being in the dict here?

File: darwin/torch/transforms.py
Comment (dorfmanrobert) [inline darwin/torch/transforms.py:1]: what do you think abt moving the "area" logic outside the logic of boxes and masks in `_post_process()`? might make things cleaner

File: darwin/torch/transforms.py [hunk 4]
```diff
@@ -378,7 +399,13 @@ def _post_process(self, albumentation_output: dict, annotation: dict) -> tuple:
             else:
                 output_annotation["masks"] = torch.stack(masks)
             if "area" in annotation:
-                output_annotation["area"] = torch.sum(output_annotation["masks"], dim=[1, 2])
+                output_annotation["area"] = torch.sum(
+                    output_annotation["masks"], dim=[1, 2]
+                )
+        elif "masks" in annotation:
```
Comment (dorfmanrobert) [inline darwin/torch/transforms.py:405]: what is this checking for/why is it needed?

File: darwin/torch/transforms.py
Comment (dorfmanrobert) [reply darwin/torch/transforms.py:1]: think the logic is this currently:

```
If "area" in annotation:
	if bboxes is not None and "masks" not in albumentation_output:
		if len(bboxes) > 0:
			output_annotation["area"] = (
                        output_annotation["boxes"][:, 2] * output_annotation["boxes"][:, 3]
                    )
		else:
			output_annotation["area"] = []
		
	elif masks is not None:
		output_annotation["area"] = torch.sum(
                    output_annotation["masks"], dim=[1, 2]
                )
                
	elif "masks" in annotation:
		output_annotation["area"] = []
```

File: darwin/torch/transforms.py (hunk 2 above)
Comment (ChristofferEdlund) [reply darwin/torch/transforms.py:366]: Python evaluates from left to right, since it is a AND and not an OR it will check if it is None and then return false (since it does not need to check the statement to the right).

File: darwin/torch/transforms.py (hunk 3 above)
Comment (ChristofferEdlund) [reply darwin/torch/transforms.py:382]: If masks are present, area calculated from the masks are more accurate then an area from bounding boxes.

File: darwin/torch/transforms.py
Comment (ChristofferEdlund) [reply darwin/torch/transforms.py:1]: Great, i like it!

File: darwin/torch/transforms.py (hunk 2 above)
Comment (dorfmanrobert) [reply darwin/torch/transforms.py:366]: oh nice good to know!

File: darwin/torch/transforms.py (hunk 4 above)
Comment (ChristofferEdlund) [reply darwin/torch/transforms.py:393]: We needed to remove the masks tag to get around an albumentation bug, but we add it back again with an empty list if there are no annotations. This is so any code on the user side that assumes there to be a mask key wont break. (It might break for empty masks, but at least not a keyerror).

File: darwin/torch/transforms.py [hunk 5]
```diff
@@ -377,8 +394,20 @@ def _post_process(self, albumentation_output: dict, annotation: dict) -> tuple:
                 output_annotation["masks"] = torch.tensor(np.array(masks))
             else:
                 output_annotation["masks"] = torch.stack(masks)
-            if "area" in annotation:
-                output_annotation["area"] = torch.sum(output_annotation["masks"], dim=[1, 2])
+        elif "masks" in annotation:
+            output_annotation["masks"] = torch.tensor([])
+
+        if "area" in annotation:
+            if "masks" in output_annotation and output_annotation["masks"].numel() > 0:
```
Comment (dorfmanrobert) [inline darwin/torch/transforms.py:401]: stylistic choice: could use `masks` and `bboxes` variables since have them but up to you

**Comment by dorfmanrobert** [comment]
left one stylistic comment abt using existing variables or not that I leave up to you, otherwise all good!

File: darwin/torch/transforms.py (hunk 5 above)
Comment (ChristofferEdlund) [reply darwin/torch/transforms.py:401]: I think we keep at as it is, masks and bboxes are not the same since we do some conversion.  masks for example is a list, and does not have .numel
